core/src/apps/common/cbor.py

```diff
@@ -184,10 +191,19 @@ def _cbor_decode(cbor: bytes) -> Tuple[Value, bytes]:
         return res, data
     elif fb_type == _CBOR_TAG:
         val, data = _read_length(cbor[1:], fb_aux)
-        item, data = _cbor_decode(data)
-        if val == _CBOR_RAW_TAG:  # only tag 24 (0x18) is supported
+        if val == _CBOR_RAW_TAG:
+            item, data = _cbor_decode(data)
             return item, data
+        elif val == _CBOR_SET_TAG:
+            aux = data[0] & _CBOR_INFO_BITS
+            ln, data = _read_length(data[1:], aux)
```

Comment:
It is customary to use `len` for length if that is what `ln` means.

Reply:
I would agree, however `ln` is used in the whole file and I am not sure what happens if my `len` would conflict with the `len()` function.